i3-nagbar/main.c

```diff
@@ -314,14 +310,63 @@ static xcb_rectangle_t get_window_position(void) {
         goto free_resources;
     }
 
-    result.x = crtc->x;
-    result.y = crtc->y;
+    result->x = crtc->x;
+    result->y = crtc->y;
     goto free_resources;
 
 free_resources:
     free(res);
     free(primary);
-    return result;
+}
+
+/**
+ * Tries to position the rectangle on the output with input focus.
+ * If unsuccessful, try to position on primary output.
+ */
+static void set_window_position_focus(xcb_rectangle_t *result) {
+    bool error = false;
```

Comment:
```suggestion
    bool success = false;
```
Instead of having multiple `error` assignments, just have a `success` one and ...